platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -186,7 +186,7 @@ class ConfigurationCacheProblems(
     }
 
     override fun forTask(task: Task): ProblemsListener {
-        val degradationReasons = degradationController.degradationReasons[task]
+        val degradationReasons = degradationController.taskDegradationReasons[task]
```

Comment:
🚀 This should be a direct lookup instead of a lookup on a temporary map of tasks

Reply:
Done